extensions/harvest/src/listTimeEntries.tsx

```diff
@@ -23,68 +26,291 @@ import {
   ToastStyle,
 } from "@raycast/api";
 import { useEffect, useState } from "react";
-import { getMyTimeEntries, restartTimer, stopTimer, useCompany } from "./services/harvest";
+import {
+  deleteTimeEntry,
+  getMyTimeEntries,
+  newTimeEntry,
+  restartTimer,
+  stopTimer,
+  useCompany,
+} from "./services/harvest";
 import { HarvestTimeEntry } from "./services/responseTypes";
 import New from "./new";
-import Delete from "./delete";
 import { execSync } from "child_process";
+import { NewTimeEntryDuration, NewTimeEntryStartEnd } from "./services/requestTypes";
+import _ from "lodash";
+
+import dayjs from "dayjs";
+import relativeTime from "dayjs/plugin/relativeTime";
+import isToday from "dayjs/plugin/isToday";
+import isTomorrow from "dayjs/plugin/isTomorrow";
+import isYesterday from "dayjs/plugin/isYesterday";
+import { AxiosError } from "axios";
+dayjs.extend(isToday);
+dayjs.extend(isTomorrow);
+dayjs.extend(isYesterday);
+dayjs.extend(relativeTime);
+
+export interface Preferences {
+  sortBy: "updated-desc" | "updated-asc" | "created-desc" | "created-asc" | "none";
+}
 
 export default function Command() {
   const [items, setItems] = useState<HarvestTimeEntry[]>([]);
   const [isLoading, setIsLoading] = useState(true);
-  // const [harvestInstallPath, setHarvestInstallPath] = useState<Application>();
+  const [viewDate, setViewDate] = useState(new Date());
+  const [navigationTitle, setNavigationTitle] = useState("Today's Timesheet");
+  const [navSubtitle, setNavSubtitle] = useState("");
+  const { data: company } = useCompany();
+
+  async function init(date?: Date) {
+    if (!date) date = viewDate;
+    let timeEntries: HarvestTimeEntry[] = [];
+
+    try {
+      timeEntries = await getMyTimeEntries({ from: date, to: date });
+    } catch (error: any) {
+      if (error.isAxiosError) {
+        if (error.response?.status === 401) {
+          await showToast(
+            ToastStyle.Failure,
+            "Invalid Token",
+            "Your API token or Account ID is invalid. Go to Raycast Preferences to update it."
+          );
+        }
+      } else {
+        await showToast(ToastStyle.Failure, "Unknown Error", "Could not fetch time entries");
+      }
```

Comment:
Shouldn't it be
```suggestion
      if (error.isAxiosError) {
        if (error.response?.status === 401) {
          await showToast(
            ToastStyle.Failure,
            "Invalid Token",
            "Your API token or Account ID is invalid. Go to Raycast Preferences to update it."
          );
        } else {
          await showToast(ToastStyle.Failure, "Unknown Error", "Could not fetch time entries");
        }
      } else {
        await showToast(ToastStyle.Failure, "Unknown Error", "Could not fetch time entries");
      }
```
so that a 500 or whatever also shows a toast? We might want to show the error message as well (if any, I'm not familiar with the API)

Reply:
Yes, good catch. I'll add that